### ISSUE TITLE
Custom Scopes

### ISSUE THREAD

**Issue by blowmage** [GoogleCloudPlatform/gcloud-ruby#209]
Each service authentication requires a different scope. Right now those scopes are hard coded, but they can be passed in when creating the service objects. We need to extend the API to allow for `:scope` to be specified.

```ruby
require "gcloud"

gcloud = Gcloud.new
pubsub = gcloud.pubsub scope: ["https://www.googleapis.com/auth/pubsub.viewer"]
```

**Comment by blowmage** [comment]
Currently for Pub/Sub and BigQuery we are asserting both a service specific and a universal access scope. We really only need to assert one, [not both](https://github.com/GoogleCloudPlatform/gcloud-ruby/blob/7ac2e988462587bb0bf503131fb6d01e69dbb7fb/lib/gcloud/pubsub/credentials.rb#L23-L24).

**Comment by blowmage** [comment]
Closed in 54bf387de52f89652d7fd9423db1734091194b0b
